### PR TITLE
Tc 670

### PR DESCRIPTION
**Pull request by hassan-yonomi**
Ability to deep sync related devices from cloud and configure it to sync on connect event or on commission event

### CODE REVIEW

**Review by ghost — APPROVED** [review]

File: src/thincloud-node-sdk.js [hunk 1]
```diff
@@ -195,20 +204,47 @@ class Client {
...
+    }
+  }
+
   publish(topic, payload, opts, cb) {
     return new Promise((resolve, reject) => {
       if (topic === '' || topic === null) {
-        log.error({ eventType: 'publish', topic: topic, data: payload });
+        log.error({
+          eventType: 'publish',
+          topic: topic,
```
Comment (mikelittman) [inline src/thincloud-node-sdk.js]: This can just be `topic` instead of `topic: topic`

File: src/thincloud-node-sdk.js [hunk 2]
```diff
@@ -195,20 +204,47 @@ class Client {
...
+        });
         throw new Error('publishFailure', 'publishFailure');
       }
 
       this._self.publish(topic, payload, opts || null, err => {
         if (err) {
-          log.error({ eventType: 'publish', topic: topic, data: payload });
+          log.error({
+            eventType: 'publish',
+            topic: topic,
```
Comment (mikelittman) [inline src/thincloud-node-sdk.js]: Same here.

File: src/thincloud-node-sdk.js [hunk 3]
```diff
@@ -195,20 +204,47 @@ class Client {
...
+          });
           reject(err);
           throw new Error('publishFailure', 'publishFailure');
         } else {
-          log.info({ eventType: 'publish', topic: topic, data: payload, opts: opts });
+          log.info({
+            eventType: 'publish',
+            topic: topic,
+            data: payload,
+            opts: opts
```
Comment (mikelittman) [inline src/thincloud-node-sdk.js]: And here with `opts` and `topic`.

**Review by mikelittman — APPROVED** [review]
Looks great!

**Pull request merged**